app/models/submission.rb

```diff
@@ -65,4 +63,12 @@ def planning_portal?
   def planx?
     source == "PlanX"
   end
+
+  private
+
+  def set_source
+    if (source = request_body.dig("metadata", "source"))
+      self.source = source
+    end
```

Comment:
Digging into the request body runs the risk of triggering an exception - is there a reason why PlanX can't include a query parameter in their request?